src/Equinox.SqlStreamStore/SqlStreamStore.fs

```diff
@@ -134,7 +134,7 @@ module private Write =
     let private writeEventsAsync (log : ILogger) (conn : IEventStoreConnection) (streamName : string) (version : int64) (events : EventData[])
         : Async<EsSyncResult> = async {
         try
-            let! wr = conn.AppendToStream(StreamId streamName, int version, events) |> Async.AwaitTaskCorrect
+            let! wr = conn.AppendToStream(StreamId streamName, (if version = -1L then ExpectedVersion.NoStream else int version), events) |> Async.AwaitTaskCorrect
```

Comment:
Actual fix for the issue.

Reply:
Cool - trusting it doesnt break MySql and Postgres ;)